DefaultSqlQueryBuilder/Extensions/SqlClauseExtensions.cs

```diff
@@ -63,10 +63,11 @@ public static IEnumerable<SqlClause> ConsolidateWhereClauses(this IEnumerable<Sq
 				.Cast<WhereSqlClause>()
 				.ToArray();
 
-			if (removableWhereClauses.Any())
+			if (removableWhereClauses.Count() > 1)
```

Comment:
The aggregate is identical to the original when there is only one clause so we skip this step.